### PR TITLE
feat: set CHE_NAMESPACE and CHE_CRD_OBJECT_NAME env vars

### PR DESCRIPTION
**Pull request by dkwon17**
Signed-off-by: David Kwon <dakwon@redhat.com>

### What does this PR do?
Sets `CHE_NAMESPACE` and `CHE_CRD_OBJECT_NAME` vars for the che-dashboard deployment.
This is done to use to use the [ServerConfigApi](https://github.com/eclipse-che/che-dashboard/blob/d1462c8457eedda63655772303ddfee599756f42/packages/dashboard-backend/src/devworkspace-client/services/api/serverConfigApi.ts#L24-L25) from che-dashboard.

The second commit adds the `server.workspacesDefaultPlugins` field in the Che CR. For example:
```
  server:
    workspacesDefaultPlugins:
      - editor: 'eclipse/che-theia/latest'
        plugins: ['https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/containers/buildah/latest/devfile.yaml']
```

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20090

### How to test this PR?
**To deploy:**
Can be tested by following these steps: https://github.com/eclipse-che/che-operator#deploy-che-operator-with-chectl-using---installer-olm-flag

I have tested this PR on CRC with this catalog image: [quay.io/dkwon17/eclipse-che-openshift-opm-catalog:next](quay.io/dkwon17/eclipse-che-openshift-opm-catalog:next).

Example:
```
chectl server:deploy --installer=olm --platform=openshift --catalog-source-yaml ./mycatalog.yaml --olm-channel=next --package-manifest-name=eclipse-che-preview-openshift
```
**To test:**
**1. Checking the che-dashboard pod's environment variables:**
image

**2. Checking the CheCluster CR for the new field:**
image

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
I don't think adding anything regarding workspces into a CheCluster CR is a good idea.

**Comment by dkwon17** [comment]
Also, why are fields regarding workspaces in the Che CR not a good idea? I think I am missing some context.

**Comment by ibuziuk** [comment]
> I don't think adding anything regarding workspces into a CheCluster CR is a good idea.

@tolusha could you please clarify why do you think that this is a bad idea? 
We have discussed the design proposal with devex team cc: @l0rd

File: pkg/deploy/dashboard/deployment_dashboard.go [hunk 1]
```diff
@@ -56,6 +56,12 @@ func (d *DashboardReconciler) getDashboardDeploymentSpec(ctx *deploy.DeployConte
 		corev1.EnvVar{
 			Name:  "CHE_URL",
 			Value: util.GetCheURL(ctx.CheCluster)},
+		corev1.EnvVar{
+			Name:  "CHE_NAMESPACE",
+			Value: ctx.CheCluster.Namespace},
```
Comment (tolusha) [inline pkg/deploy/dashboard/deployment_dashboard.go:61]: BTW, namespace can be read from file `/var/run/secrets/kubernetes.io/serviceaccount/namespace`.

File: pkg/deploy/dashboard/deployment_dashboard.go [hunk 2]
```diff
@@ -56,6 +56,12 @@ func (d *DashboardReconciler) getDashboardDeploymentSpec(ctx *deploy.DeployConte
...
+		corev1.EnvVar{
+			Name:  "CHE_CRD_OBJECT_NAME",
```
Comment (tolusha) [inline pkg/deploy/dashboard/deployment_dashboard.go:63]: Better name `CHECLUSTER_CR_NAME`.
If you decide to mount the env with a namspace name, then be consistent in naming. I suggest `CHECLUSTER_CR_NAMESPACE`.

**Review by ibuziuk — APPROVED** [review]

**Comment by dkwon17** [comment]
Thank you for the feedback @tolusha , I made some changes

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
pls merge

**Pull request merged**